### PR TITLE
CompGCN

### PR DESCRIPTION
**Pull request by YijianLiu**
## Description
Finish CompGcn(pytorch) for entity classification

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/compgcn/comgcn_trainer.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+def main(args):
+    # load cora dataset
+    if str.lower(args.dataset) not in ['aifb', 'mutag', 'bgs', 'am']:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    path = osp.join(osp.dirname(osp.realpath(__file__)), '../Entities')
+    dataset = Entities(path, args.dataset)
+    graph = dataset[0]
+
+    graph.numpy()
```
Comment (dddg617) [inline examples/compgcn/comgcn_trainer.py:68]: Why we need to convert to numpy()? In line 71, you convert to tensor()?

File: examples/compgcn/comgcn_trainer.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    best_val_acc = 0
+    for epoch in range(args.n_epoch):
+        net.set_train()
+        train_loss = train_one_step(data, train_y)
+        net.set_eval()
+        logits = net(data['edge_index'], data['edge_type'])
+        val_logits = tlx.gather(logits, data['test_idx'])
+        val_acc = calculate_acc(val_logits, data['test_y'], metrics)
```
Comment (dddg617) [inline examples/compgcn/comgcn_trainer.py:139]: Why the variable is valid accuracy but we use ``test`` in the input parameters.

File: examples/rgcn/readme.md [hunk 3]
```diff
@@ -18,13 +18,13 @@ Results
...
 TL_BACKEND="pytorch" python rgcn_trainer.py --dataset mutag --l2_coef 5e-2
-TL_BACKEND="pytorch" python rgcn_trainer.py --dataset bgs --lr 0.0001 --l2_coef 5e-2
+
 
 TL_BACKEND="tensorflow" python rgcn_trainer.py --dataset aifb 
 TL_BACKEND="tensorflow" python rgcn_trainer.py --dataset mutag --l2_coef 5e-2
 TL_BACKEND="tensorflow" python rgcn_trainer.py --dataset bgs --l2_coef 5e-2
 
-TL_BACKEND="paddle" python rgcn_trainer.py --dataset aifb 
+TL_BACKEND="paddle" python rgcn_traineTL_BACKEND="pytorch" python rgcn_trainer.py --dataset bgs --lr 0.0001 --l2_coef 5e-2r.py --dataset aifb
```
Comment (dddg617) [inline examples/rgcn/readme.md:27]: I think we do not need to change this file.

File: gammagl/layers/conv/compgcn_conv.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        Input a and b should have the same dimensions. And this operation supports broadcasting.
+        Returns
+        -------
+        Tensor, having the same dimension as the input a.
+        """
+
+        def com_mult(a, b):
+            r1, i1 = a[..., 0], a[..., 1]
+            r2, i2 = b[..., 0], b[..., 1]
+            return torch.stack([r1 * r2 - i1 * i2, r1 * i2 + i1 * r2], dim=-1)
```
Comment (dddg617) [inline gammagl/layers/conv/compgcn_conv.py:175]: Do not use torch in the ``conv`` layer. Change to ``tensorlayerx``.

File: gammagl/layers/conv/compgcn_conv.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        try:
+            from torch import irfft
+            from torch import rfft
+        except ImportError:
+            from torch.fft import irfft2
+            from torch.fft import rfft2
+
+            def rfft(x, d):
+                t = rfft2(x, dim=(-d))
+                return torch.stack((t.real, t.imag), -1)
```
Comment (dddg617) [inline gammagl/layers/conv/compgcn_conv.py:190]: Ditto.

File: gammagl/layers/conv/compgcn_conv.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+            def irfft(x, d, signal_sizes):
+                return irfft2(torch.complex(x[:, :, 0], x[:, :, 1]), s=signal_sizes, dim=(-d))
```
Comment (dddg617) [inline gammagl/layers/conv/compgcn_conv.py:193]: Ditto.

File: gammagl/layers/conv/compgcn_conv.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        dst_index = edge_index[0, :]
+        if aggr == 'sum':
+            return unsorted_segment_sum(msg, dst_index, num_nodes)
+            #return unsorted_segment_sum(msg, dst_index, num_nodes)
+        elif aggr == 'mean':
+            return unsorted_segment_mean(msg, dst_index, num_nodes)
+        elif aggr == 'max':
+            return unsorted_segment_max(msg, dst_index, num_nodes)
+        else:
+            raise NotImplementedError('Not support for this opearator')
```
Comment (dddg617) [inline gammagl/layers/conv/compgcn_conv.py:154]: This function is of no use, delete this.

File: examples/compgcn/comgcn_trainer.py (hunk 1 above)
Comment (YijianLiu) [reply examples/compgcn/comgcn_trainer.py:68]: I have deleted them.

File: examples/compgcn/comgcn_trainer.py (hunk 2 above)
Comment (YijianLiu) [reply examples/compgcn/comgcn_trainer.py:139]: hhhh, the paradigm of train and val is what I copy from RGCN(ggl), maybe it doesn't have valid dataset, what about we ask RGCN together.

File: examples/rgcn/readme.md (hunk 3 above)
Comment (YijianLiu) [reply examples/rgcn/readme.md:27]: yes

File: gammagl/layers/conv/compgcn_conv.py (hunk 4 above)
Comment (YijianLiu) [reply gammagl/layers/conv/compgcn_conv.py:175]: All these are used for fast fourier transform, but tensorlaye doesn't have it. I think we can temporarily use pytorch
